routing/joint_index.cpp

```diff
@@ -4,11 +4,21 @@
 
 namespace routing
 {
+RoadPoint JointIndex::GetPoint(Joint::Id jointId) const
+{
+  if (IsStatic(jointId))
+    return m_points[Begin(jointId)];
+
+  CHECK_LESS(jointId, GetNumJoints(), ());
+  auto const jointIt = m_dynamicJoints.find(jointId);
+  CHECK(jointIt != m_dynamicJoints.cend(), ());
+  return jointIt->second.GetEntry(0 /* first point in joint */);
+}
+
 Joint::Id JointIndex::InsertJoint(RoadPoint const & rp)
 {
   Joint::Id const jointId = GetNumJoints();
-  m_points.emplace_back(rp);
-  m_offsets.emplace_back(m_points.size());
+  AppendToJoint(jointId, rp);
```

Comment:
The joint id was increased in GetNumJoints(). GetNumJoints() returned size of static joints and dynamic joints. And the size of |m_dynamicJoints| was increased in AppendToJoint(). But anyway now the method is refactored.